build.gradle

```diff
@@ -51,7 +51,9 @@ subprojects {
     dependencies {
         implementation 'com.google.guava:guava:31.0.1-jre'
         implementation platform('io.micrometer:micrometer-bom:1.7.5')
+        implementation 'org.apache.logging.log4j:log4j-core:2.15.0'
```

Comment:
I do not believe that we should have `log4` in the subprojects. This should only be in Data Prepper core.

Reply:
This code block configures dependencies for all sub-projects. So any dependency here is used in all sub-projects.

Data Prepper uses SLF4J which provides a logging abstraction. All the sub-projects need SLF4J. Indeed it is included here:

```
implementation 'org.slf4j:slf4j-api:1.7.32'
```

However, only the application needs to include an actual SLF4J implementation. The application is `data-prepper-core` and it uses Log4j as the SLF4J implementation.

This is where data-prepper-core includes Log4j: https://github.com/opensearch-project/data-prepper/blob/main/data-prepper-core/build.gradle#L31-L32

--

Basically, lines 54 and 56 are unnecessary and should just be deleted.